realm/realm-library/src/main/java/io/realm/Realm.java

```diff
@@ -1630,9 +1640,21 @@ public void delete(Class<? extends RealmModel> clazz) {
 
 
     @SuppressWarnings("unchecked")
-    private <E extends RealmModel> E copyOrUpdate(E object, boolean update, Map<RealmModel, RealmObjectProxy> cache) {
+    private <E extends RealmModel> E copyOrUpdate(E object, boolean update, Map<RealmModel, RealmObjectProxy> cache, Set<ImportFlag> flags) {
         checkIfValid();
-        return configuration.getSchemaMediator().copyOrUpdate(this, object, update, cache);
+        if (!isInTransaction()) {
```

Comment:
are you sure this check is not done in the public/caller methods?

Reply:
Not sure exactly what you mean. This is the public method? And this isn't verified at the lower levels anymore as we are no longer using the setters of the `Row`.